hack/verify-no-vendor-cycles.sh

```diff
@@ -34,21 +34,8 @@ staging_repos_pattern=$(IFS="|"; echo "${staging_repos[*]}")
 
 cd "${KUBE_ROOT}"
 
```

Comment:
this will create and clean up a temp dir for you:
```
kube::util::ensure-temp-dir()
go list -mod=vendor -test -deps -json ./vendor/... > "${KUBE_TEMP}/deps.json"
```